modules/claims_api/app/services/claims_api/bgs_to_lighthouse_claims_mapper_service.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  class BGSToLighthouseClaimsMapperService < ClaimsApi::Service
+    attr_accessor :bgs_claims, :lighthouse_claims
+
+    def initialize(bgs_claims:, lighthouse_claims:)
+      @bgs_claims        = bgs_claims
+      @lighthouse_claims = lighthouse_claims
+    end
+
+    def process
+      return [] unless claims_exist?
+
+      mapped_claims = map_claims
+      mapped_claims = append_remaining_lighthouse_claims(mapped_claims: mapped_claims) if lighthouse_claims.present?
+      mapped_claims
+    end
+
+    private
+
+    def claims_exist?
+      return true if bgs_claims.key?(:bnft_claim_detail) || lighthouse_claims.present?
+
+      false
+    end
+
+    def map_claims
+      bgs_claims[:bnft_claim_detail].map do |bgs_claim|
+        if find_bgs_claim_in_lighthouse_collection(claim: bgs_claim)
+          matching_claim = find_bgs_claim_in_lighthouse_collection(claim: bgs_claim)
```

Comment:
nit: if this is moved above the `if` check the logic in this method wouldn't have to be executed twice

Reply:
Fair enough.
[Changed here](https://github.com/department-of-veterans-affairs/vets-api/pull/7395/commits/7bd5e8be206124ab757b40ae285b01e945f331a0)